test/loader/BasePackageLoader.test.ts

```diff
@@ -0,0 +1,1390 @@
+        url: 'http://example.org/fhir/locals/StructureDefinition/human-being-logical-model',
+        name: 'Human',
+        version: '1.0.0',
+        sdKind: 'logical',
+        sdDerivation: 'specialization',
+        sdType: 'http://example.org/fhir/locals/StructureDefinition/human-being-logical-model',
+        sdBaseDefinition: 'http://hl7.org/fhir/StructureDefinition/Base',
+        sdAbstract: false,
+        sdCharacteristics: ['can-be-target'],
+        sdFlavor: 'Logical',
+        packageName: 'human-being-logical-model',
+        packageVersion: '1.0.0',
+        resourcePath: humanBeingResourcePath
+      });
+      expect(result).toBe(LoadStatus.LOADED);
+    });
+
+    it('should use LOCAL package paths to find package', async () => {
+      const humanBeingResourcePath = path.resolve(
+        __dirname,
+        'fixtures',
+        'StructureDefinition-human-being-logical-model.json'
+      );
+      const pkg = setupLoadPackage(
+        'LOCAL',
+        'LOCAL',
+        'not-loaded',
+        undefined,
+        path.resolve(__dirname, 'fixtures'),
+        humanBeingResourcePath,
+        '2024-05-24T16:27:17-04:00'
+      );
+      packageCacheMock.isPackageInCache.calledWith(pkg.name, pkg.version).mockReturnValue(true);
+      packageCacheMock.getPackagePath
+        .calledWith(pkg.name, pkg.version)
+        .mockReturnValue(pkg.packageJsonPath);
+      packageCacheMock.getPotentialResourcePaths
+        .calledWith(pkg.name, pkg.version)
+        .mockReturnValue([pkg.resourcePath]);
+      const humanBeingJSON = fs.readJsonSync(humanBeingResourcePath);
+      packageCacheMock.getResourceAtPath
+        .calledWith(pkg.resourcePath)
+        .mockReturnValue(humanBeingJSON);
+      packageDBMock.getPackageStats
+        .calledWith(pkg.name, pkg.version)
+        .mockReturnValue({ name: pkg.name, version: pkg.version, resourceCount: 1 });
+
+      const result = await loader.loadPackage('LOCAL', 'LOCAL');
+      expect(packageDBMock.savePackageInfo).toHaveBeenCalled();
+      expect(packageDBMock.saveResourceInfo).toHaveBeenCalledWith({
+        resourceType: 'StructureDefinition',
+        id: 'human-being-logical-model',
+        url: 'http://example.org/fhir/locals/StructureDefinition/human-being-logical-model',
+        name: 'Human',
+        version: '1.0.0',
+        sdKind: 'logical',
+        sdDerivation: 'specialization',
+        sdType: 'http://example.org/fhir/locals/StructureDefinition/human-being-logical-model',
+        sdBaseDefinition: 'http://hl7.org/fhir/StructureDefinition/Base',
+        sdAbstract: false,
+        sdCharacteristics: ['can-be-target'],
+        sdFlavor: 'Logical',
+        packageName: 'LOCAL',
+        packageVersion: 'LOCAL',
+        resourcePath: humanBeingResourcePath
+      });
+      expect(result).toBe(LoadStatus.LOADED);
+    });
+
+    it('should find a logical that uses the logical-target extension to set its characteristics', async () => {
```

Comment:
It seems like there are a series of tests here whose basic intent is to test the code in the private `loadResource` method to ensure that all the ResourceInfo is extracted and loaded correctly. That being the case, it seems there is probably a lot of duplicate code across these tests. Is there an opportunity to create a common test function that basically takes in a resource path and the expected ResourceInfo that should be produced upon loading?